packages/atomic/src/components/search/atomic-facet-manager/atomic-facet-manager.tsx

```diff
@@ -95,12 +95,13 @@ export class AtomicFacetManager implements InitializableComponent {
   }
 
   private get facets() {
-    const facets: BaseFacetElement[] = [];
     const children = Array.from(this.host.children);
 
-    children.forEach((child) => {
-      this.isPseudoFacet(child) && facets.push(child);
-    });
+    const facets = children.filter(
+      (child) =>
+        this.isPseudoFacet(child) ||
+        child.tagName === 'ATOMIC-AUTOMATIC-FACET-BUILDER'
+    ) as BaseFacetElement[];
```

Comment:
Array.prototype.filter is a generic function in typescript, meaning  you could do:
```suggestion
    const facets = children.filter<BaseFacetElement>(
      (child) =>
        this.isPseudoFacet(child) ||
        child.tagName === 'ATOMIC-AUTOMATIC-FACET-BUILDER'
    );
```

You may need to explicitly declare the return type of your predicate so that it is a typeguard func

Reply:
This doenst work. There needs to be a cast somewhere.